src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1282,7 +1284,7 @@ public int SendTo(byte[] buffer, int offset, int size, SocketFlags socketFlags,
             Internals.SocketAddress socketAddress = Serialize(ref remoteEP);
 
             int bytesTransferred;
-            SocketError errorCode = SocketPal.SendTo(_handle, buffer, offset, size, socketFlags, socketAddress.Buffer, socketAddress.Size, out bytesTransferred);
+            SocketError errorCode = SocketPal.SendTo(_handle, buffer, offset, size, socketFlags, socketAddress.Buffer, out bytesTransferred);
```

Comment:
Do we need to slice socketAddress.Buffer to socketAddress.Size?

Reply:
That is done by the SocketAddress e.g. it returns internal buffer up to the Size. (e.g. it does not show whole underlying buffer). Without it, it would be hard to use IMHO as everybody would need to keep slicing.

Reply:
According to this line, it's sliced to InternalSize, not Size:
https://github.com/dotnet/runtime/pull/88970/files#diff-840f5a24cac9941cc101bdffdd4772c982b00fe9b71e2aa8aa48a95746d4fa96R161
Can't those be different?

Reply:
Perhaps I misunderstood the question during review. Size can be sliced down to be less than buffer size but effectively it is `InternalSize`. (e.g. InternalSize holds the visible/valid part of the InternalBuffer)

https://github.com/dotnet/runtime/blob/8f70860b23845e7b6a2fc31c1edf699d3ee2935d/src/libraries/Common/src/System/Net/SocketAddress.cs#L47-L53

When SocketAddress is created for given `AddressFamily`  `Size`/`InternalSize` is set to appropriate length and the buffer length is exactly that (as we do not Rent from pool) Somebody can check that or use the new `GetMaximumAddressSize`

We can make it to return always the full buffer if you feel it is better. It may be more work to get the correct data (e.g. Slicing) and Im not sure if there is value getting access to parts that do not hold valid content. 
This really matters only for Receiving, right?

Reply:
> InternalSize holds the visible/valid part of the InternalBuffer

Ok. It's confusing that we have both Size and InternalSize, when they're actually the same thing (I thought they were different).  We should rename `InternalSize` to be `_size`.

> We can make it to return always the full buffer if you feel it is better. It may be more work to get the correct data (e.g. Slicing) and Im not sure if there is value getting access to parts that do not hold valid content.
This really matters only for Receiving, right?

I don't have a super strong opinion.  If you feel that what you have is better and will lead to more folks falling into the pit of success, I'm ok with it.  We should then be clear in the documentation that Buffer.Length always equals Size but the internal buffer may actually be larger, and if you want to set Size, you can set it up to the value it originally was when the SocketAddress was constructed.

Reply:
I decided to keep the Buffer sliced to current address to make it easier to use. The only one case IMHO where this may be tricky is SocketAddress constructive when somebody would explicit put smaller size than needed. I added test and it would fail with strange address error on Windows and it did hit some assets in Socket PAL on Linux. So I added chart to make sure passed SA is always good enough. This should always be the case for internal use in Sockets. I will rename InternalSize in follow-up PR.